pkg/healthcheck/workloads.go

```diff
@@ -0,0 +1,245 @@
+package healthcheck
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/openshift/microshift/pkg/config"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/cmd/get"
+	"k8s.io/kubectl/pkg/cmd/rollout"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/utils/ptr"
+)
+
+type NamespaceWorkloads struct {
+	Deployments  []string `json:"deployments"`
+	DaemonSets   []string `json:"daemonsets"`
+	StatefulSets []string `json:"statefulsets"`
+}
+
+func waitForWorkloads(timeout time.Duration, workloads map[string]NamespaceWorkloads) error {
+	aeg := &AllErrGroup{}
+	for ns, wls := range workloads {
+		aeg.Go(func() error { return waitForWorkloadsInNamespace(timeout, ns, wls) })
+	}
+
+	errs := aeg.Wait()
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+func getCoreMicroShiftWorkloads() (map[string]NamespaceWorkloads, error) {
+	cfg, err := config.ActiveConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	workloads := map[string]NamespaceWorkloads{
+		"openshift-ovn-kubernetes": {
+			DaemonSets: []string{"ovnkube-master", "ovnkube-node"},
+		},
+		"openshift-service-ca": {
+			Deployments: []string{"service-ca"},
+		},
+		"openshift-ingress": {
+			Deployments: []string{"router-default"},
+		},
+		"openshift-dns": {
+			DaemonSets: []string{
+				"dns-default",
+				"node-resolver",
+			},
+		},
+	}
+	fillOptionalWorkloadsIfApplicable(cfg, workloads)
+
+	return workloads, nil
+}
```

Comment:
I'm a bit concerned that this information is "hidden" deep in the code. 
Should we cancel the default list and provide it all explicitly from greenboot?

Reply:
The idea was that `microshift healthcheck` is something that works standalone, without greenboot.
It can be executed via greenboot scripts, but can be without.
This would break that assumption and make `microshift healthcheck` useless for running the command manually.

Reply:
Currently, the information on the checked namespaces is hidden. We could still leave the option to run healthcheck on core namespaces by default, but keep this information external to be loaded and used by the command.